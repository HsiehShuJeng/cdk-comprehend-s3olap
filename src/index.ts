export * from './cdk-comprehend-s3olap';
export * from './comprehend-lambdas';
export * from './iam-roles';