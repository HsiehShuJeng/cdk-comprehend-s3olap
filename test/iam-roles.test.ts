import * as cdk from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { GeneralRole, AdminRole, BillingRole, CustSupportRole } from '../src/iam-roles';

test('IAM roles test', () => {
  const app = new cdk.App();
  const stack = new cdk.Stack(app, 'DummyStack');
  new GeneralRole(stack, 'GeneralRole', {});
  new AdminRole(stack, 'AdminRole');
  new BillingRole(stack, 'BillingRole');
  new CustSupportRole(stack, 'CustSupportRole');
  const template = Template.fromStack(stack);

  template.resourceCountIs('AWS::IAM::Role', 4);
  template.hasResourceProperties('AWS::IAM::Role', {
    RoleName: 'GeneralRole',
    AssumeRolePolicyDocument: {
      Statement: [
        {
          Action: 'sts:AssumeRole',
          Effect: 'Allow',
          Principal: {
            AWS: {
              'Fn::Join': [
                '',
                [
                  'arn:',
                  {
                    Ref: 'AWS::Partition',
                  },
                  ':iam::',
                  {
                    Ref: 'AWS::AccountId',
                  },
                  ':root',
                ],
              ],
            },
          },
        },
      ],
      Version: '2012-10-17',
    },
    Policies: [
      {
        PolicyDocument: {
          Statement: [
            {
              Action: 's3:ListBucket',
              Effect: 'Allow',
              Resource: '*',
              Sid: 'AllowListingObjects',
            },
            {
              Action: [
                's3:GetAccessPointForObjectLambda',
                's3:GetAccessPointConfigurationForObjectLambda',
                's3:ListAccessPointsForObjectLambda',
                's3:ListAllMyBuckets',
                's3:ListAccessPoints',
                's3:GetAccessPoint',
                's3:GetAccountPublicAccessBlock',
                's3:GetBucketPublicAccessBlock',
                's3:GetBucketPolicyStatus',
                's3:GetBucketAcl',
                's3:GetAccessPointPolicyStatus',
              ],
              Effect: 'Allow',
              Resource: '*',
              Sid: 'AllowListingBucketsAndAccessPoints',
            },
            {
              Action: [
                's3-object-lambda:Get*',
                's3-object-lambda:List*',
              ],
              Effect: 'Allow',
              Resource: {
                'Fn::Join': [
                  '',
                  [
                    'arn:',
                    {
                      Ref: 'AWS::Partition',
                    },
                    ':s3-object-lambda:',
                    {
                      Ref: 'AWS::Region',
                    },
                    ':',
                    {
                      Ref: 'AWS::AccountId',
                    },
                    ':accesspoint/accessctl-s3olap-survey-results-unknown-pii',
                  ],
                ],
              },
              Sid: 'AllowObjectLambdaAccess',
            },
            {
              Action: [
                's3:Get*',
                's3:List*',
              ],
              Condition: {
                'ForAnyValue:StringEquals': {
                  'aws:CalledVia': [
                    's3-object-lambda.amazonaws.com',
                  ],
                },
              },
              Effect: 'Allow',
              Resource: '*',
              Sid: 'AllowStandardAccessPointAccess',
            },
            {
              Action: 'lambda:InvokeFunction',
              Condition: {
                'ForAnyValue:StringEquals': {
                  'aws:CalledVia': [
                    's3-object-lambda.amazonaws.com',
                  ],
                },
              },
              Effect: 'Allow',
              Resource: '*',
              Sid: 'AllowLambdaInvocation',
            },
          ],
          Version: '2012-10-17',
        },
        PolicyName: 'general-role-s3olap-policy',
      },
    ],
  });
  template.hasResourceProperties('AWS::IAM::Role', {
    RoleName: 'RedactionAdminRole',
    AssumeRolePolicyDocument: {
      Statement: [
        {
          Action: 'sts:AssumeRole',
          Effect: 'Allow',
          Principal: {
            AWS: {
              'Fn::Join': [
                '',
                [
                  'arn:',
                  {
                    Ref: 'AWS::Partition',
                  },
                  ':iam::',
                  {
                    Ref: 'AWS::AccountId',
                  },
                  ':root',
                ],
              ],
            },
          },
        },
      ],
      Version: '2012-10-17',
    },
    Policies: [
      {
        PolicyDocument: {
          Statement: [
            {
              Action: 's3:ListBucket',
              Effect: 'Allow',
              Resource: '*',
              Sid: 'AllowListingObjects',
            },
            {
              Action: [
                's3:GetAccessPointForObjectLambda',
                's3:GetAccessPointConfigurationForObjectLambda',
                's3:ListAccessPointsForObjectLambda',
                's3:ListAllMyBuckets',
                's3:ListAccessPoints',
                's3:GetAccessPoint',
                's3:GetAccountPublicAccessBlock',
                's3:GetBucketPublicAccessBlock',
                's3:GetBucketPolicyStatus',
                's3:GetBucketAcl',
                's3:GetAccessPointPolicyStatus',
              ],
              Effect: 'Allow',
              Resource: '*',
              Sid: 'AllowListingBucketsAndAccessPoints',
            },
            {
              Action: [
                's3-object-lambda:Get*',
                's3-object-lambda:List*',
              ],
              Effect: 'Allow',
              Resource: {
                'Fn::Join': [
                  '',
                  [
                    'arn:',
                    {
                      Ref: 'AWS::Partition',
                    },
                    ':s3-object-lambda:',
                    {
                      Ref: 'AWS::Region',
                    },
                    ':',
                    {
                      Ref: 'AWS::AccountId',
                    },
                    ':accesspoint/admin-s3olap-call-transcripts-known-pii',
                  ],
                ],
              },
              Sid: 'AllowObjectLambdaAccess',
            },
            {
              Action: [
                's3:Get*',
                's3:List*',
              ],
              Condition: {
                'ForAnyValue:StringEquals': {
                  'aws:CalledVia': [
                    's3-object-lambda.amazonaws.com',
                  ],
                },
              },
              Effect: 'Allow',
              Resource: '*',
              Sid: 'AllowStandardAccessPointAccess',
            },
            {
              Action: 'lambda:InvokeFunction',
              Condition: {
                'ForAnyValue:StringEquals': {
                  'aws:CalledVia': [
                    's3-object-lambda.amazonaws.com',
                  ],
                },
              },
              Effect: 'Allow',
              Resource: '*',
              Sid: 'AllowLambdaInvocation',
            },
          ],
          Version: '2012-10-17',
        },
        PolicyName: 'admin-role-s3olap-policy',
      },
    ],
  });
  template.hasResourceProperties('AWS::IAM::Role', {
    RoleName: 'RedactionBillingRole',
    AssumeRolePolicyDocument: {
      Statement: [
        {
          Action: 'sts:AssumeRole',
          Effect: 'Allow',
          Principal: {
            AWS: {
              'Fn::Join': [
                '',
                [
                  'arn:',
                  {
                    Ref: 'AWS::Partition',
                  },
                  ':iam::',
                  {
                    Ref: 'AWS::AccountId',
                  },
                  ':root',
                ],
              ],
            },
          },
        },
      ],
      Version: '2012-10-17',
    },
    Policies: [
      {
        PolicyDocument: {
          Statement: [
            {
              Action: 's3:ListBucket',
              Effect: 'Allow',
              Resource: '*',
              Sid: 'AllowListingObjects',
            },
            {
              Action: [
                's3:GetAccessPointForObjectLambda',
                's3:GetAccessPointConfigurationForObjectLambda',
                's3:ListAccessPointsForObjectLambda',
                's3:ListAllMyBuckets',
                's3:ListAccessPoints',
                's3:GetAccessPoint',
                's3:GetAccountPublicAccessBlock',
                's3:GetBucketPublicAccessBlock',
                's3:GetBucketPolicyStatus',
                's3:GetBucketAcl',
                's3:GetAccessPointPolicyStatus',
              ],
              Effect: 'Allow',
              Resource: '*',
              Sid: 'AllowListingBucketsAndAccessPoints',
            },
            {
              Action: [
                's3-object-lambda:Get*',
                's3-object-lambda:List*',
              ],
              Effect: 'Allow',
              Resource: {
                'Fn::Join': [
                  '',
                  [
                    'arn:',
                    {
                      Ref: 'AWS::Partition',
                    },
                    ':s3-object-lambda:',
                    {
                      Ref: 'AWS::Region',
                    },
                    ':',
                    {
                      Ref: 'AWS::AccountId',
                    },
                    ':accesspoint/billing-s3olap-call-transcripts-known-pii',
                  ],
                ],
              },
              Sid: 'AllowObjectLambdaAccess',
            },
            {
              Action: [
                's3:Get*',
                's3:List*',
              ],
              Condition: {
                'ForAnyValue:StringEquals': {
                  'aws:CalledVia': [
                    's3-object-lambda.amazonaws.com',
                  ],
                },
              },
              Effect: 'Allow',
              Resource: '*',
              Sid: 'AllowStandardAccessPointAccess',
            },
            {
              Action: 'lambda:InvokeFunction',
              Condition: {
                'ForAnyValue:StringEquals': {
                  'aws:CalledVia': [
                    's3-object-lambda.amazonaws.com',
                  ],
                },
              },
              Effect: 'Allow',
              Resource: '*',
              Sid: 'AllowLambdaInvocation',
            },
          ],
          Version: '2012-10-17',
        },
        PolicyName: 'billing-role-s3olap-policy',
      },
    ],
  });
  template.hasResourceProperties('AWS::IAM::Role', {
    RoleName: 'RedactionCustSupportRole',
    AssumeRolePolicyDocument: {
      Statement: [
        {
          Action: 'sts:AssumeRole',
          Effect: 'Allow',
          Principal: {
            AWS: {
              'Fn::Join': [
                '',
                [
                  'arn:',
                  {
                    Ref: 'AWS::Partition',
                  },
                  ':iam::',
                  {
                    Ref: 'AWS::AccountId',
                  },
                  ':root',
                ],
              ],
            },
          },
        },
      ],
      Version: '2012-10-17',
    },
    Policies: [
      {
        PolicyDocument: {
          Statement: [
            {
              Action: 's3:ListBucket',
              Effect: 'Allow',
              Resource: '*',
              Sid: 'AllowListingObjects',
            },
            {
              Action: [
                's3:GetAccessPointForObjectLambda',
                's3:GetAccessPointConfigurationForObjectLambda',
                's3:ListAccessPointsForObjectLambda',
                's3:ListAllMyBuckets',
                's3:ListAccessPoints',
                's3:GetAccessPoint',
                's3:GetAccountPublicAccessBlock',
                's3:GetBucketPublicAccessBlock',
                's3:GetBucketPolicyStatus',
                's3:GetBucketAcl',
                's3:GetAccessPointPolicyStatus',
              ],
              Effect: 'Allow',
              Resource: '*',
              Sid: 'AllowListingBucketsAndAccessPoints',
            },
            {
              Action: [
                's3-object-lambda:Get*',
                's3-object-lambda:List*',
              ],
              Effect: 'Allow',
              Resource: {
                'Fn::Join': [
                  '',
                  [
                    'arn:',
                    {
                      Ref: 'AWS::Partition',
                    },
                    ':s3-object-lambda:',
                    {
                      Ref: 'AWS::Region',
                    },
                    ':',
                    {
                      Ref: 'AWS::AccountId',
                    },
                    ':accesspoint/custsupport-s3olap-call-transcripts-known-pii',
                  ],
                ],
              },
              Sid: 'AllowObjectLambdaAccess',
            },
            {
              Action: [
                's3:Get*',
                's3:List*',
              ],
              Condition: {
                'ForAnyValue:StringEquals': {
                  'aws:CalledVia': [
                    's3-object-lambda.amazonaws.com',
                  ],
                },
              },
              Effect: 'Allow',
              Resource: '*',
              Sid: 'AllowStandardAccessPointAccess',
            },
            {
              Action: 'lambda:InvokeFunction',
              Condition: {
                'ForAnyValue:StringEquals': {
                  'aws:CalledVia': [
                    's3-object-lambda.amazonaws.com',
                  ],
                },
              },
              Effect: 'Allow',
              Resource: '*',
              Sid: 'AllowLambdaInvocation',
            },
          ],
          Version: '2012-10-17',
        },
        PolicyName: 'customersupport-role-s3olap-policy',
      },
    ],
  });
});