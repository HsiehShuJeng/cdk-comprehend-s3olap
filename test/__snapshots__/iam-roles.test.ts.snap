// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IAM roles test 1`] = `
Object {
  "Resources": Object {
    "AdminRoleRedactionAdminRole0DE28314": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Policies": Array [
          Object {
            "PolicyDocument": Object {
              "Statement": Array [
                Object {
                  "Action": "s3:ListBucket",
                  "Effect": "Allow",
                  "Resource": "*",
                  "Sid": "AllowListingObjects",
                },
                Object {
                  "Action": Array [
                    "s3:GetAccessPointForObjectLambda",
                    "s3:GetAccessPointConfigurationForObjectLambda",
                    "s3:ListAccessPointsForObjectLambda",
                    "s3:ListAllMyBuckets",
                    "s3:ListAccessPoints",
                    "s3:GetAccessPoint",
                    "s3:GetAccountPublicAccessBlock",
                    "s3:GetBucketPublicAccessBlock",
                    "s3:GetBucketPolicyStatus",
                    "s3:GetBucketAcl",
                    "s3:GetAccessPointPolicyStatus",
                  ],
                  "Effect": "Allow",
                  "Resource": "*",
                  "Sid": "AllowListingBucketsAndAccessPoints",
                },
                Object {
                  "Action": Array [
                    "s3-object-lambda:Get*",
                    "s3-object-lambda:List*",
                  ],
                  "Effect": "Allow",
                  "Resource": Object {
                    "Fn::Join": Array [
                      "",
                      Array [
                        "arn:",
                        Object {
                          "Ref": "AWS::Partition",
                        },
                        ":s3-object-lambda:",
                        Object {
                          "Ref": "AWS::Region",
                        },
                        ":",
                        Object {
                          "Ref": "AWS::AccountId",
                        },
                        ":accesspoint/admin-s3olap-call-transcripts-known-pii",
                      ],
                    ],
                  },
                  "Sid": "AllowObjectLambdaAccess",
                },
                Object {
                  "Action": Array [
                    "s3:Get*",
                    "s3:List*",
                  ],
                  "Condition": Object {
                    "ForAnyValue:StringEquals": Object {
                      "aws:CalledVia": Array [
                        "s3-object-lambda.amazonaws.com",
                      ],
                    },
                  },
                  "Effect": "Allow",
                  "Resource": "*",
                  "Sid": "AllowStandardAccessPointAccess",
                },
                Object {
                  "Action": "lambda:InvokeFunction",
                  "Condition": Object {
                    "ForAnyValue:StringEquals": Object {
                      "aws:CalledVia": Array [
                        "s3-object-lambda.amazonaws.com",
                      ],
                    },
                  },
                  "Effect": "Allow",
                  "Resource": "*",
                  "Sid": "AllowLambdaInvocation",
                },
              ],
              "Version": "2012-10-17",
            },
            "PolicyName": "admin-role-s3olap-policy",
          },
        ],
        "RoleName": "RedactionAdminRole",
      },
      "Type": "AWS::IAM::Role",
    },
    "BillingRoleRedactionBillingRole72FB0886": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Policies": Array [
          Object {
            "PolicyDocument": Object {
              "Statement": Array [
                Object {
                  "Action": "s3:ListBucket",
                  "Effect": "Allow",
                  "Resource": "*",
                  "Sid": "AllowListingObjects",
                },
                Object {
                  "Action": Array [
                    "s3:GetAccessPointForObjectLambda",
                    "s3:GetAccessPointConfigurationForObjectLambda",
                    "s3:ListAccessPointsForObjectLambda",
                    "s3:ListAllMyBuckets",
                    "s3:ListAccessPoints",
                    "s3:GetAccessPoint",
                    "s3:GetAccountPublicAccessBlock",
                    "s3:GetBucketPublicAccessBlock",
                    "s3:GetBucketPolicyStatus",
                    "s3:GetBucketAcl",
                    "s3:GetAccessPointPolicyStatus",
                  ],
                  "Effect": "Allow",
                  "Resource": "*",
                  "Sid": "AllowListingBucketsAndAccessPoints",
                },
                Object {
                  "Action": Array [
                    "s3-object-lambda:Get*",
                    "s3-object-lambda:List*",
                  ],
                  "Effect": "Allow",
                  "Resource": Object {
                    "Fn::Join": Array [
                      "",
                      Array [
                        "arn:",
                        Object {
                          "Ref": "AWS::Partition",
                        },
                        ":s3-object-lambda:",
                        Object {
                          "Ref": "AWS::Region",
                        },
                        ":",
                        Object {
                          "Ref": "AWS::AccountId",
                        },
                        ":accesspoint/billing-s3olap-call-transcripts-known-pii",
                      ],
                    ],
                  },
                  "Sid": "AllowObjectLambdaAccess",
                },
                Object {
                  "Action": Array [
                    "s3:Get*",
                    "s3:List*",
                  ],
                  "Condition": Object {
                    "ForAnyValue:StringEquals": Object {
                      "aws:CalledVia": Array [
                        "s3-object-lambda.amazonaws.com",
                      ],
                    },
                  },
                  "Effect": "Allow",
                  "Resource": "*",
                  "Sid": "AllowStandardAccessPointAccess",
                },
                Object {
                  "Action": "lambda:InvokeFunction",
                  "Condition": Object {
                    "ForAnyValue:StringEquals": Object {
                      "aws:CalledVia": Array [
                        "s3-object-lambda.amazonaws.com",
                      ],
                    },
                  },
                  "Effect": "Allow",
                  "Resource": "*",
                  "Sid": "AllowLambdaInvocation",
                },
              ],
              "Version": "2012-10-17",
            },
            "PolicyName": "billing-role-s3olap-policy",
          },
        ],
        "RoleName": "RedactionBillingRole",
      },
      "Type": "AWS::IAM::Role",
    },
    "CustSupportRoleRedactionCustSupportRole3F19038E": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Policies": Array [
          Object {
            "PolicyDocument": Object {
              "Statement": Array [
                Object {
                  "Action": "s3:ListBucket",
                  "Effect": "Allow",
                  "Resource": "*",
                  "Sid": "AllowListingObjects",
                },
                Object {
                  "Action": Array [
                    "s3:GetAccessPointForObjectLambda",
                    "s3:GetAccessPointConfigurationForObjectLambda",
                    "s3:ListAccessPointsForObjectLambda",
                    "s3:ListAllMyBuckets",
                    "s3:ListAccessPoints",
                    "s3:GetAccessPoint",
                    "s3:GetAccountPublicAccessBlock",
                    "s3:GetBucketPublicAccessBlock",
                    "s3:GetBucketPolicyStatus",
                    "s3:GetBucketAcl",
                    "s3:GetAccessPointPolicyStatus",
                  ],
                  "Effect": "Allow",
                  "Resource": "*",
                  "Sid": "AllowListingBucketsAndAccessPoints",
                },
                Object {
                  "Action": Array [
                    "s3-object-lambda:Get*",
                    "s3-object-lambda:List*",
                  ],
                  "Effect": "Allow",
                  "Resource": Object {
                    "Fn::Join": Array [
                      "",
                      Array [
                        "arn:",
                        Object {
                          "Ref": "AWS::Partition",
                        },
                        ":s3-object-lambda:",
                        Object {
                          "Ref": "AWS::Region",
                        },
                        ":",
                        Object {
                          "Ref": "AWS::AccountId",
                        },
                        ":accesspoint/custsupport-s3olap-call-transcripts-known-pii",
                      ],
                    ],
                  },
                  "Sid": "AllowObjectLambdaAccess",
                },
                Object {
                  "Action": Array [
                    "s3:Get*",
                    "s3:List*",
                  ],
                  "Condition": Object {
                    "ForAnyValue:StringEquals": Object {
                      "aws:CalledVia": Array [
                        "s3-object-lambda.amazonaws.com",
                      ],
                    },
                  },
                  "Effect": "Allow",
                  "Resource": "*",
                  "Sid": "AllowStandardAccessPointAccess",
                },
                Object {
                  "Action": "lambda:InvokeFunction",
                  "Condition": Object {
                    "ForAnyValue:StringEquals": Object {
                      "aws:CalledVia": Array [
                        "s3-object-lambda.amazonaws.com",
                      ],
                    },
                  },
                  "Effect": "Allow",
                  "Resource": "*",
                  "Sid": "AllowLambdaInvocation",
                },
              ],
              "Version": "2012-10-17",
            },
            "PolicyName": "customersupport-role-s3olap-policy",
          },
        ],
        "RoleName": "RedactionCustSupportRole",
      },
      "Type": "AWS::IAM::Role",
    },
    "GeneralRoleIamRoleD3D7532B": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "AWS": Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      "arn:",
                      Object {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      Object {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Policies": Array [
          Object {
            "PolicyDocument": Object {
              "Statement": Array [
                Object {
                  "Action": "s3:ListBucket",
                  "Effect": "Allow",
                  "Resource": "*",
                  "Sid": "AllowListingObjects",
                },
                Object {
                  "Action": Array [
                    "s3:GetAccessPointForObjectLambda",
                    "s3:GetAccessPointConfigurationForObjectLambda",
                    "s3:ListAccessPointsForObjectLambda",
                    "s3:ListAllMyBuckets",
                    "s3:ListAccessPoints",
                    "s3:GetAccessPoint",
                    "s3:GetAccountPublicAccessBlock",
                    "s3:GetBucketPublicAccessBlock",
                    "s3:GetBucketPolicyStatus",
                    "s3:GetBucketAcl",
                    "s3:GetAccessPointPolicyStatus",
                  ],
                  "Effect": "Allow",
                  "Resource": "*",
                  "Sid": "AllowListingBucketsAndAccessPoints",
                },
                Object {
                  "Action": Array [
                    "s3-object-lambda:Get*",
                    "s3-object-lambda:List*",
                  ],
                  "Effect": "Allow",
                  "Resource": Object {
                    "Fn::Join": Array [
                      "",
                      Array [
                        "arn:",
                        Object {
                          "Ref": "AWS::Partition",
                        },
                        ":s3-object-lambda:",
                        Object {
                          "Ref": "AWS::Region",
                        },
                        ":",
                        Object {
                          "Ref": "AWS::AccountId",
                        },
                        ":accesspoint/accessctl-s3olap-survey-results-unknown-pii",
                      ],
                    ],
                  },
                  "Sid": "AllowObjectLambdaAccess",
                },
                Object {
                  "Action": Array [
                    "s3:Get*",
                    "s3:List*",
                  ],
                  "Condition": Object {
                    "ForAnyValue:StringEquals": Object {
                      "aws:CalledVia": Array [
                        "s3-object-lambda.amazonaws.com",
                      ],
                    },
                  },
                  "Effect": "Allow",
                  "Resource": "*",
                  "Sid": "AllowStandardAccessPointAccess",
                },
                Object {
                  "Action": "lambda:InvokeFunction",
                  "Condition": Object {
                    "ForAnyValue:StringEquals": Object {
                      "aws:CalledVia": Array [
                        "s3-object-lambda.amazonaws.com",
                      ],
                    },
                  },
                  "Effect": "Allow",
                  "Resource": "*",
                  "Sid": "AllowLambdaInvocation",
                },
              ],
              "Version": "2012-10-17",
            },
            "PolicyName": "general-role-s3olap-policy",
          },
        ],
        "RoleName": "GeneralRole",
      },
      "Type": "AWS::IAM::Role",
    },
  },
}
`;
